packages/vc-verification/src/issuer-def-resolver.ts

```diff
@@ -0,0 +1,44 @@
+import { IIssuerDefinition } from '@energyweb/credential-governance';
+import { providers } from 'ethers';
+import { RoleDefinitionResolverV2__factory } from '@energyweb/credential-governance/ethers/factories/RoleDefinitionResolverV2__factory';
+import { RoleDefinitionResolverV2 } from '@energyweb/credential-governance/ethers/RoleDefinitionResolverV2';
+
+/**
+ * An interface for Resolution of issuers for a namespace
+ */
+export interface IssuerDefinitionResolver {
+  /**
+   * @param namespace
+   * @returns IIssuerDefinition for the given namespace
+   */
+  getRoleIssuerDefinition(namespace: string): Promise<IIssuerDefinition>;
+}
+
+export class RoleIssuerDefinitionResolver implements IssuerDefinitionResolver {
```

Comment:
IMO, this name could be improved. I would add something about how it is resolving using an ethers provider (to contrast it with a resolver from cached data, for example). Name is bit clunky but I think 
```suggestion
/**
 * Resolves an issuer definition by reading smart contract state via an Ethers provider
 */
export class EthersProviderIssuerDefinitionResolver implements IssuerDefinitionResolver {
```